### ISSUE TITLE
Deepar Notebook - Different outputs for different runs despite setting the seed.

### ISSUE THREAD

**Issue by misclassified** [awslabs/amazon-sagemaker-examples#1058]
Hi,

I've observed a problem of instability with the DeepAr algorithm as of deepar-electricity notebook. Basically when running twice, despite setting both a numpy seed and random seed, the error output changes, often even largely.

The issue can be reproduced by running the notebook twice and taking note of the cross-validation out using `estimator.training_job_analytics.dataframe()`

How is that possible?

**Comment by larroy** [comment]
Probably you need to also set the mxnet seed.

**Comment by larroy** [comment]
@lostella

**Comment by vafl** [comment]
I assume that you set the seeds locally in your notebook, in this case the seeds are not set for the algorithm on sagemaker. Currently there is no way to provide seeds for DeepAR training job. In general there is always some variance when training deep learning algorithms. You can perhaps reduce the variance for these runs, by changing some hyper parameters (e.g. training for more epochs and reducing the learning rate).

**Comment by misclassified** [comment]
thanks for the quick reply. In the notebook they set the seed, but considering what you say this is totally irrelevant and can be misleading, I would recommend removing that.  

Not being able to set the seed can be irrelevant for a home project, but in commercial application getting two different outputs when running the same data, the same algorithm, the same parameters twice, is pretty bad, as results are not reproducible and not auditable. 

I'm not sure tweaking the learning rate helps, maybe setting the context length to a small number helps, but this is contrary to what the authors recommend with this algorithm. 

Is there a way to request this change for future releases?

**Comment by larroy** [comment]
@vafl why additional random seeds would DeepAR need that can't be set through np, mxnet...

**Comment by lostella** [comment]
@larroy the seeds on the instance that runs the algorithm, which one cannot control

**Comment by larroy** [comment]
@lostella I got it, makes sense.
